### PR TITLE
*: make transaction duration metrics more reasonable

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
The transaction duration metrics seem not right for the users. The users may want an end to end duration.

### What is changed and how it works?
Make it more reasonable.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. run some workload
1. check out the transaction duration metrics

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the `tidb-ansible` repository

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: session/tidb.go [hunk 1]
```diff
@@ -165,12 +165,23 @@ func Compile(ctx context.Context, sctx sessionctx.Context, stmtNode ast.StmtNode
...
+		var aborted bool
 		if !sessVars.InTxn() {
 			logutil.BgLogger().Info("rollbackTxn for ddl/autocommit failed")
 			se.RollbackTxn(ctx)
+			aborted = true
 		} else if se.txn.Valid() && se.txn.IsPessimistic() && executor.ErrDeadlock.Equal(meetsErr) {
 			logutil.BgLogger().Info("rollbackTxn for deadlock", zap.Uint64("txn", se.txn.StartTS()))
 			se.RollbackTxn(ctx)
+			aborted = true
+		}
```
Comment (zimulala) [inline session/tidb.go]: I think we can add an `else` on line177, and we return the meetsErr in this `else` statement. Then we can remove `aborted`.

Comment (jackysp) [reply session/tidb.go]: But all the branches need to `return meetErr`.

Comment (jackysp) [reply session/tidb.go]: But all the branches need to `return meetErr`.

Comment (jackysp) [reply session/tidb.go]: PTAL thx

**Pull request merged**